src/test/test-execute.c

```diff
@@ -1288,8 +1288,9 @@ static int prepare_ns(const char *process_name) {
         if (r == 0) {
                 _cleanup_free_ char *unit_dir = NULL;
 
-                /* Make "/" read-only. */
-                assert_se(mount_nofollow_verbose(LOG_DEBUG, NULL, "/", NULL, MS_BIND|MS_REMOUNT, NULL) >= 0);
+                /* Make "/" and "/root" read-only. */
+                FOREACH_STRING(p, "/", "/root")
+                        assert_se(mount_nofollow_verbose(LOG_DEBUG, NULL, p, NULL, MS_BIND|MS_REMOUNT, NULL) >= 0);
```

Comment:
I may be confused, but instead of remounting /root/ here, I guess we should do that after MS_SHARED|MS_REC mount below?

Reply:
Yeah, sorry about that. I was about to leave and didn't really test this at all, lemme rework this.